src/main/java/edu/kis/powp/jobs2d/drivers/adapter/StraightLineAdapter.java

```diff
@@ -1,18 +1,20 @@
 package edu.kis.powp.jobs2d.drivers.adapter;
 
-import edu.kis.legacy.drawer.panel.DrawPanelController;
+ import edu.kis.legacy.drawer.panel.DrawPanelController;
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class StraightLineAdapter implements Job2dDriver {
 	private int startX = 0, startY = 0;
-
-	public MyAdapter() {
+	private DrawPanelController drawController;
+	public StraightLineAdapter() {
 		super();
+		drawController = DrawerFeature.getDrawerController();
```

Comment:
Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.